onnxruntime/core/session/inference_session.cc

```diff
@@ -946,6 +971,46 @@ class InferenceSession::Impl {
     session_state_.SetLogger(*session_logger_);
   }
 
+  // Registers all the predefined transformers with transformer manager
+  void AddPredefinedTransformers(GraphTransformerManager& transformer_manager, const uint32_t& enabled_levels, const std::vector<std::string>& custom_list) {
+    if ((enabled_levels & TransformerLevel::Optional_L1) || !custom_list.empty()) {
+      auto level = TransformerLevel::Optional_L1;
+      // Generate and register rewrite rules for L1
+      auto rewrite_rules_to_register = transformerutils::GenerateRewriteRules(level, custom_list);
+      if (!rewrite_rules_to_register.empty()) {
+        std::unique_ptr<RuleBasedGraphTransformer> graph_rewrite_rules = std::make_unique<TopDownRuleBasedTransformer>("L1_RuleBasedTransformer", "Top down transformer");
+        for (auto& entry : rewrite_rules_to_register) {
+          graph_rewrite_rules->Register(std::move(entry));
+        }
+        transformer_manager.Register(std::move(graph_rewrite_rules), level, {"", onnxruntime::kCpuExecutionProvider});
+      }
+
+      // Generate and register transformers for L1
+      auto transformers_to_register = transformerutils::GenerateTransformers(level, custom_list);
+      for (auto& entry : transformers_to_register) {
+        transformer_manager.Register(std::move(entry.first), level, std::move(entry.second));
+      }
+    }
+
+    if ((enabled_levels & TransformerLevel::Optional_L2) || !custom_list.empty()) {
+      auto level = TransformerLevel::Optional_L2;
+      // Generate and register rewrite rules for L2
+      auto rewrite_rules_to_register = transformerutils::GenerateRewriteRules(level, custom_list);
+      if (!rewrite_rules_to_register.empty()) {
+        std::unique_ptr<RuleBasedGraphTransformer> graph_rewrite_rules = std::make_unique<TopDownRuleBasedTransformer>("L2_RuleBasedTransformer", "Top down transformer");
+        for (auto& entry : rewrite_rules_to_register) {
+          graph_rewrite_rules->Register(std::move(entry));
+        }
+        transformer_manager.Register(std::move(graph_rewrite_rules), level, {onnxruntime::kCpuExecutionProvider});
+      }
+
+      auto transformers_to_register = transformerutils::GenerateTransformers(level, custom_list);
+      for (auto& entry : transformers_to_register) {
+        transformer_manager.Register(std::move(entry.first), level, std::move(entry.second));
+      }
+    }
+  }  
```

Comment:
This looks like a cut-and-paste of the above for L1. Can you instead put the logic in a lambda and just pass in a level as a param?